iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -125,39 +124,32 @@ private enum StateFlags
             _deviceBoundOneWayProcessor = new SimpleWorkQueue<PublishPacket>(AcceptMessageAsync);
             _deviceBoundTwoWayProcessor = new OrderedTwoPhaseWorkQueue<int, PublishPacket>(AcceptMessageAsync, p => p.PacketId, SendAckAsync);
 
-            _serviceBoundOneWayProcessor = new SimpleWorkQueue<PublishWorkItem>(this.SendMessageToServerAsync);
+            _serviceBoundOneWayProcessor = new SimpleWorkQueue<PublishWorkItem>(SendMessageToServerAsync);
             _serviceBoundTwoWayProcessor = new OrderedTwoPhaseWorkQueue<int, PublishWorkItem>(SendMessageToServerAsync, p => p.Value.PacketId, ProcessAckAsync);
         }
 
+        private int InboundBacklogSize => _deviceBoundOneWayProcessor.BacklogSize + _deviceBoundTwoWayProcessor.BacklogSize;
+
         #region IChannelHandler overrides
 
-        public override async void ChannelActive(IChannelHandlerContext context)
+        public override void ChannelActive(IChannelHandlerContext context)
         {
-            if (Logging.IsEnabled)
-            {
-                Logging.Enter(this, context.Name, nameof(ChannelActive));
-            }
+            Logging.Enter(this, context.Name, nameof(ChannelActive));
 
             _stateFlags = StateFlags.NotConnected;
 
-            await ConnectAsync(context).ConfigureAwait(true);
```

Comment:
is this to avoid fire-and-forget of async void?

Reply:
Yes